src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -124,28 +124,32 @@ public TypeSymbolWithAnnotations GetInferredReturnType(ConversionsBase conversio
         /// <summary>
         /// Behavior of this function should be kept aligned with <see cref="UnboundLambdaState.ReturnInferenceCacheKey"/>.
         /// </summary>
-        internal static InferredLambdaReturnType InferReturnType(ArrayBuilder<(RefKind, TypeSymbolWithAnnotations)> returnTypes, CSharpCompilation compilation, ConversionsBase conversions, TypeSymbol delegateType, bool isAsync)
+        internal static InferredLambdaReturnType InferReturnType(ArrayBuilder<(BoundReturnStatement, TypeSymbolWithAnnotations)> returnTypes,
+            BoundNode node, CSharpCompilation compilation, ConversionsBase conversions, TypeSymbol delegateType, bool isAsync)
```

Comment:
Is it possible to use more specific type for the ```node```?

Reply:
The main purpose for this node is to provide syntax to get a binder from. It is also stored in `AbstractFlowPass` base (and used in `Scan` method, which I don't think `NullableWalker` uses).
This is the existing parameter name from `NullableWalker` constructors.